research-software-directory/RSD-as-a-service
# 1207 fix: add support for mathml in mentions

# Support mathml stynax in the publication titles (mentions)

Closes #1207

Changes proposed in this pull request:
*  New dependency added [dompurify](https://github.com/cure53/DOMPurify) to sanitize "scraped" html 
*  Created SanitizedMathMLBox to use to sanitize received data. It uses MUI Box component which makes possible to use any valid html element. **We allow only MathML elements (see tests)**.
*  Related change: limit mention title to 5 lines and show complete title on mouseover
*  Related change: limited authors height to 9rem, and add vertical scrollbar when content is larger

How to test:
* `make start` to build
* login as rsd-admin and create new software item 'Test software 1'
* Edit software, navigate to mentions and import mentions `10.1103/physrevd.108.072003` 
* Search for publication "test". Select the one with very long title "TEST TEST TEST ..."
* Confirm that MathML is readable on the software page
* Confirm that authors list is also fully visible (it will be scrollable list) taking 9rem of vertical space

# Example edit software
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/ebced9cc-901f-4272-b5ca-82f62d10cd97)

# Example view software
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/9a6fcb70-99f2-4be9-b253-242424ed17d8)


PR Checklist:
